From 34f0b9ff977e0af278fb0ed3ef3bf9ad9fa87ff9 Mon Sep 17 00:00:00 2001
From: Vladimir Ermakov <vooon341@gmail.com>
Date: Tue, 17 Jun 2014 01:14:26 +0400
Subject: [PATCH 4/7] Add cmake rules for pymavlink installation (v2).

Use external script. (as catkin).
Some code from catkin build system used to fix installation process.
---
 pymavlink/CMakeLists.txt                 | 23 +++++++++++++++++++++++
 pymavlink/python_distutils_install.sh.in | 22 ++++++++++++++++++++++
 2 files changed, 45 insertions(+)
 create mode 100644 pymavlink/CMakeLists.txt
 create mode 100755 pymavlink/python_distutils_install.sh.in

diff --git a/pymavlink/CMakeLists.txt b/pymavlink/CMakeLists.txt
new file mode 100644
index 0000000..7fb27ed
--- /dev/null
+++ b/pymavlink/CMakeLists.txt
@@ -0,0 +1,23 @@
+# pymavlink support's python2 only
+set(PythonInterp_FIND_VERSION "2")
+find_package(PythonInterp REQUIRED)
+
+# part of catkin/python.cmake
+set(enable_setuptools_deb_layout OFF)
+if(EXISTS "/etc/debian_version")
+  set(enable_setuptools_deb_layout ON)
+endif()
+option(SETUPTOOLS_DEB_LAYOUT "Enable debian style python package layout" ${enable_setuptools_deb_layout})
+
+if(SETUPTOOLS_DEB_LAYOUT)
+  message(STATUS "Using Debian Python package layout")
+  set(SETUPTOOLS_ARG_EXTRA "--install-layout=deb")
+else()
+  message(STATUS "Using default Python package layout")
+endif()
+
+# prepare installation script
+set(INSTALL_SCRIPT ${CMAKE_CURRENT_BINARY_DIR}/python_distutils_install.sh)
+configure_file(python_distutils_install.sh.in ${INSTALL_SCRIPT} @ONLY)
+
+install(CODE "execute_process(COMMAND \"${INSTALL_SCRIPT}\" WORKING_DIRECTORY \"${CMAKE_CURRENT_SOURCE_DIR}\")")
diff --git a/pymavlink/python_distutils_install.sh.in b/pymavlink/python_distutils_install.sh.in
new file mode 100755
index 0000000..b39c2ed
--- /dev/null
+++ b/pymavlink/python_distutils_install.sh.in
@@ -0,0 +1,22 @@
+#!/bin/sh -x
+# Modified version of catkin template/python_distutils_install.sh.in
+
+if [ -n "$DESTDIR" ] ; then
+    case $DESTDIR in
+        /*) # ok
+            ;;
+        *)
+            /bin/echo "DESTDIR argument must be absolute... "
+            /bin/echo "otherwise python's distutils will bork things."
+            exit 1
+    esac
+    DESTDIR_ARG="--root=$DESTDIR"
+fi
+
+/usr/bin/env \
+    "@PYTHON_EXECUTABLE@" \
+    "@CMAKE_CURRENT_SOURCE_DIR@/setup.py" \
+    build --build-base "@CMAKE_CURRENT_BINARY_DIR@/pybuild" \
+    install \
+    $DESTDIR_ARG \
+    @SETUPTOOLS_ARG_EXTRA@ --prefix="@CMAKE_INSTALL_PREFIX@"
-- 
1.9.1

